### ISSUE TITLE
Using std::chrono::steady_clock instead of  system_clock in RaftKeeper

### ISSUE THREAD

**Issue by lzydmxy** [JDRaftKeeper/RaftKeeper#266]
### Description

The following is flame graph of RaftKeeper instance which serving as ClickHouse metadata store. A lot of cpu time cost () on  std::chrono::system::clock (because of syscall).Maybe we could optimize the std::chrono::steady_clock  instead of  std::chrono::steady_clock
![image](https://github.com/JDRaftKeeper/RaftKeeper/assets/49020515/36c7978b-9456-4338-ba29-59a5ce18008e)
![image](https://github.com/JDRaftKeeper/RaftKeeper/assets/49020515/7d42c68d-078e-454f-a424-f3a94957a63f)

small test
```
#include <iostream>
#include <chrono>

using steady_clock = std::chrono::steady_clock;
using system_clock = std::chrono::system_clock;

int main()
{
    auto t1 = steady_clock::now();
    auto t2 = steady_clock::now();
    auto t3 = steady_clock::now();
    std::cout << "chrono::steady_clock: " << (t3-t1).count() << "ns" << std::endl;


    auto t4 = steady_clock::now();
    auto t5 = system_clock::now();
    auto t6 = steady_clock::now();
    std::cout << "chrono::system_clock: " << (t6-t4).count() << "ns" << std::endl;
}
```
./main
chrono::steady_clock: 197ns
chrono::system_clock: 501ns

twice std::chrono::steady_clock cost 197ns
once std::chrono::steady_clock once std::chrono::system_clock cost 500ns

once std::chrono::system_clock  may cost 400ns and once std::chrono::steady_clock  may cost 100ns

**Comment by JackyWoo** [comment]
hi, I make a simple test with the following code snippet:

```
inline size_t getCurrentTimestamp()
{
    auto now = std::chrono::steady_clock::now();
    return std::chrono::duration_cast<std::chrono::nanoseconds>(now.time_since_epoch()).count();
}

int main()
{
    size_t n = 3;
    auto t1 = getCurrentTimestamp();

    for (size_t i=0; i<n; i++)
        std::chrono::steady_clock::now();

    auto t2 = getCurrentTimestamp();

    for (size_t i=0; i<n; i++)
        std::chrono::system_clock::now();

    auto t3 = getCurrentTimestamp();

    std::cout << "steady clock: " << std::to_string(t2 - t1) << "ns" << std::endl;
    std::cout << "system clock: " << std::to_string(t3 - t2) << "ns" << std::endl;
}
```

I found that the result are quiet different in multiple runing:

The 1st run:
```
steady clock: 167ns
system clock: 208ns
```

The 2nd run:
```
steady clock: 250ns
system clock: 166ns
```

The 3rd run:
```
steady clock: 291ns
system clock: 292ns
```

PS: my env is MacOS(M3) clang16

**Comment by JackyWoo** [comment]
Different result in Ubuntu 20.4

Nearly same with in multiple runnings
```
steady clock: 261ns
system clock: 577ns
```

**Comment by JackyWoo** [comment]
Will be fixed by #352
